wfl/generate/md/abort.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+class AbortBase(ABC):
+    """Base class used for checking and aborting MD simulation of `wfl.generate.md.sample()`"""
+    def __init__(self, n_failed_steps=1):
+        self.history = []
+        self.n_failed_steps = n_failed_steps
+
+    @abstractmethod
+    def check_if_atoms_ok(self, at):
```

Comment:
Why is this in the base class? Shouldn't it be up to each derived class to decide what infrastructure it needs to make the final "stop MD" decision?

Reply:
This is in the base class, but it's an abstract method, so the derived classes _must_ implement their own particular version. My reasoning was - we need to enforce a method we know a name of so that `md.sample` can call it. That is `stop_md`, which by default just stops the dynamics on the first failure, but uses the particular `check_if_atoms_ok` (better names welcome). Derived classes can of course re-implement `stop_md` too to have something more particular.

Reply:
OK, I see - the essentially the default behavior is to stop on the first failure, and the derived classes can override that by reimplementing the actual stop function. I'm fine with that, but let me tweak the docstrings so I think that behavior is clear, and I also want to change the function names.

Reply:
Fixed in the upcoming commit.